### PR TITLE
Version 311 - MSVC Fix and Extend MSVC tests

### PR DESCRIPTION
**Pull request by madmongo1**

### CODE REVIEW

**Comment by sdarwin** [comment]
@madmongo1 ,  I just noticed something else:

Immediately after each line of actual testing in the drone script, add this line, so the script will exit with an error.     That is, if the script runs 3 tests, this should  appear 3 times:
```
if %errorlevel% neq 0 exit /b %errorlevel%
```

**Comment by madmongo1** [comment]
> @madmongo1 , I just noticed something else:
> 
> Immediately after each line of actual testing in the drone script, add this line, so the script will exit with an error. That is, if the script runs 3 tests, this should appear 3 times:
> 
> ```
> if %errorlevel% neq 0 exit /b %errorlevel%
> ```

done on next push. thanks

File: include/boost/beast/core/impl/multi_buffer.hpp [hunk 1]
```diff
@@ -916,7 +916,7 @@ prepare(size_type n) ->
             std::size_t altn = in_size_ * growth_factor;
 	    // Overflow detection:
             if(in_size_ > altn)
-                altn = std::numeric_limits<std::size_t>::max();
+                altn = (std::numeric_limits<std::size_t>::max)();
```
Comment (vinniefalco) [inline include/boost/beast/core/impl/multi_buffer.hpp:919]: Maybe do a search for `"max("` (and `"min("`) in all files?

**Comment by vinniefalco** [comment]
This looks good, as long as there are no other occurrences

File: include/boost/beast/core/impl/multi_buffer.hpp (hunk 1 above)
Comment (madmongo1) [reply include/boost/beast/core/impl/multi_buffer.hpp:919]: You must be looking at an old commit?
I have 
```
            if(in_size_ > altn)
                altn = (std::numeric_limits<std::size_t>::max)();
```

Comment (madmongo1) [reply include/boost/beast/core/impl/multi_buffer.hpp:919]: Ah I see. it was just a comment.

**Pull request merged**